### DISCUSSION TITLE
Is it possible to return json body in "+page.server.js"

### DISCUSSION THREAD

**Discussion by Chetan11-dev** [sveltejs/kit#6067] [Q&A] [answered]
Goal is to return a json response from POST action "+page.server.js".

I tried returning body like this, but did not work
```js
export async function POST({ request }) {
	return {
		status: 204,
		body: {
			content: "some-content"
		}
	};
}
```

**Comment by fmaclen** [comment, marked as answer]
According to the [migration guide](https://github.com/sveltejs/kit/discussions/5774#discussioncomment-3292699):

> **+page.server.js**
> `POST`, `PUT` and `PATCH` handlers should not return data. They can return an errors object of validation errors that can be used for UI (provided to the component as an errors prop), plus an optional status that defaults to 400 when error is provided and is otherwise ignored...

I'm in the middle of migrating myself so someone else can correct me if I'm wrong but in my app I have `/my-endpoint.json/+server.js` and I submit `POST` requests to `/my-endpoint.json` getting a [`Response`](https://kit.svelte.dev/docs/web-standards#fetch-apis-response), in your case it would be something like:

```javascript
// ~/your-endpoint.json/+server.js
import { json } from '@sveltejs/kit';

export async function POST({ request }) {
  return json({
     content: "some-content"
  });
}
```
`json(...)` will return a `Response` object with a status `200`, but if you need to return a `204` I think it'd look something like this:
```javascript
return new Response(JSON.stringify({ content: "some-content" }), {
  headers: {
    'Content-Type': 'application/json'
  },
  status: 204
});
```

I say _"I think"_ because I actually get a 500 error when I set the status to `204`. Other status codes like `200`, `201`, `202` and even `206` work fine, but `204` doesn't:

```
Response constructor: Invalid response status code.
```

**Reply by Chetan11-dev** [reply]
Thank You

**Reply by adonis2611** [reply]
How will we get this data back in a +page.svelte file? 

`
+page.svelte
const send = async () => {
		const submit = await fetch('/', {
			method: 'POST',
			headers: {
				'content-type': 'application/json',
				accept: 'application/json'
			},
			body: JSON.stringify({
				name
			})
		});
		const data = await submit.json();
		console.log(data);
	};
`
`
+page.sever.js
export async function POST({ request }) {
	const { name } = await request.json();

	return json({
		content: 'some-content'
	});
}

`
